### PR TITLE
Make Developing a bit Easier

### PR DESCRIPTION
**Pull request by justindbaur**
- Ignore files that might appear on your local machine while developing
- Correct the current working directory in VSCode so that the content root is in the expected place and it reads `*.json` configuration files.
- In `Development` register services and endpoints so that if migrations have not been applied it gives you an easy button to do that.
- Add values in `appsettings.Development.json` that are values common for local environments and facilitate less setup
- Change the default location for the database to be directly in the content root since that directory is guaranteed to exist and makes applying migrations easier.

### CODE REVIEW

File: src/AdminConsole/Program.cs [hunk 1]
```diff
@@ -158,7 +162,12 @@ void RunTheApp()
 
 
     // Configure the HTTP request pipeline.
-    if (!app.Environment.IsDevelopment())
+    if (app.Environment.IsDevelopment())
+    {
+        app.UseDeveloperExceptionPage();
+        app.UseMigrationsEndPoint();
```
Comment (abergs) [inline src/AdminConsole/Program.cs:168]: Nice

File: src/AdminConsole/appsettings.Development.json [hunk 2]
```diff
@@ -11,9 +11,15 @@
...
   },
   "ConnectionStrings": {
-    "sqlite": "Data Source=Data/adminconsole_dev.db"
+    "sqlite": "Data Source=adminconsole_dev.db"
   },
   "Passwordless": {
-    "ApiUrl": "https://localhost:8001"
+    "ApiUrl": "http://localhost:7001",
+    "ApiSecret": "TestValue",
+    "ApiKey": "TestValue"
```
Comment (abergs) [inline src/AdminConsole/appsettings.Development.json:19]: Thought: could we use proper values that we seed in #30?

**Review by abergs — APPROVED** [review]
Please note that I tweaked it, feel free to review those changes :)

**Review by abergs — APPROVED** [review]
Please note that I tweaked it, feel free to review those changes :)

File: src/AdminConsole/appsettings.Development.json (hunk 2 above)
Comment (justindbaur) [reply src/AdminConsole/appsettings.Development.json:19]: Are you seeding values in #30? I don't see such a change.

File: src/AdminConsole/Services/Mail/FileMailProvider.cs [hunk 3]
```diff
@@ -6,8 +6,9 @@ public class FileMailProvider : IMailProvider
     public Task SendAsync(MailMessage message)
     {
         string msg = message.TextBody;
-        Console.WriteLine(msg);
-        File.AppendAllText("mail.txt", msg);
+        Console.WriteLine("> Sent email to mail.md");
```
Comment (justindbaur) [inline src/AdminConsole/Services/Mail/FileMailProvider.cs:9]: I think we should inject `ILogger` and use that instead.

File: src/AdminConsole/Pages/Organization/Verify.cshtml [hunk 4]
```diff
@@ -6,4 +6,14 @@
...
+<p>We've sent a confirmation link to the email address you provided. Click on the link in the email to verify your account</p>
+
+<environment names="Development">
+    @{
+        var mailPath = System.IO.Path.GetFullPath("mail.md");
+    }
+    <br />
+    <b>Development mode: </b> Please open the the <code>src/AdminConsole/mail.md</code> file.
+    <br />
+    <a class="link-blue" href="vscode://file/@mailPath">Open with VS Code: @mailPath</a>
```
Comment (justindbaur) [inline src/AdminConsole/Pages/Organization/Verify.cshtml:18]: Very cool!

File: src/AdminConsole/appsettings.Development.json (hunk 2 above)
Comment (abergs) [reply src/AdminConsole/appsettings.Development.json:19]: Not seeding in #30 currently, but if we're adding database creation / migration there, we could also add seeding

**Comment by abergs** [comment]
I'm going to tidy this up and then I think it's ready for merge, right?

**Comment by justindbaur** [comment]
@abergs Looks great to me!

**Pull request merged**